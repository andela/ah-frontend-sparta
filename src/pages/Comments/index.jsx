import React from 'react';
import Comment from '../../components/Comment';
import ReplyToArticle from '../../components/ReplyToArticle';

function structureComments(comments, showReplies, replyDisplayState, slug, postReplyToComment, deleteComment, updateComment, onChangeComment) {
  const jsxComments = [];
  for (let i = 0; i < comments.length; i += 1) {
    const comment = comments[i];
    jsxComments.push(<Comment key={comment.id} commentObject={comment} showReplies={showReplies} replyDisplayState={replyDisplayState} slug={slug} postReplyToComment={postReplyToComment} deleteComment={deleteComment} updateComment={updateComment} onChangeComment={onChangeComment} />);
  }
  return jsxComments;
}

const Comments = ({
  comments,
  showReplies,
  replyDisplayState,
  slug,
  postReplyToComment,
  onCommentChanged,
  postComment,
  deleteComment,
  updateComment,
  onChangeComment,
}) => (
  <div>
    <div style={{ marginTop: 100, marginBottom: 15 }}>
      <ReplyToArticle
        onCommentChanged={onCommentChanged}
        postComment={postComment}
      />
      <span style={{ fontSize: 28 }}>
        {comments.length}
        {' '}
        Comments
      </span>
    </div>
    {structureComments(comments, showReplies, replyDisplayState, slug, postReplyToComment, deleteComment, updateComment, onChangeComment)}
  </div>
);

export default Comments;
