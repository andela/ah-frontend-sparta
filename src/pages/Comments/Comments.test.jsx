import React from 'react';
import renderer from 'react-test-renderer';
import Comments from './index';

test('Comments snapshot test', () => {
  const props = {
    comments: [],
    showReplies: jest.fn(),
    replyDisplayState: jest.fn(),
    slug: 'comments-slug',
    postReplyToComment: jest.fn(),
    onCommentChanged: jest.fn(),
    postComment: jest.fn(),
    deleteComment: jest.fn(),
    updateComment: jest.fn(),
    onChangeComment: jest.fn(),
  };
  const wrapper = renderer.create(<Comments {...props} />);
  expect(wrapper).toMatchSnapshot();
});
