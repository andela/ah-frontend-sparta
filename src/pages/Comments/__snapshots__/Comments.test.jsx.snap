// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments snapshot test 1`] = `
<div>
  <div
    style={
      Object {
        "marginBottom": 15,
        "marginTop": 100,
      }
    }
  >
    <div
      className="form-group shadow-textarea"
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <textarea
        className="form-control commentReplyTextarea z-depth-1"
        name="comment"
        onChange={[MockFunction]}
        placeholder="Comment on Article"
        rows="5"
      />
      <button
        className="commentReplyButtn"
        onClick={[MockFunction]}
        type="submit"
      >
        POST
         
      </button>
    </div>
    <span
      style={
        Object {
          "fontSize": 28,
        }
      }
    >
      0
       
      Comments
    </span>
  </div>
</div>
`;
